From d67d1cda5f44d322059d0a2115c77038e5e88a5c Mon Sep 17 00:00:00 2001
From: Frederick Roy <fredroy@users.noreply.github.com>
Date: Tue, 26 Aug 2025 16:57:56 +0900
Subject: [PATCH] [Framework] Fix compilation with clang20 (#5665)

* fix compilation with clang20

* remove deprecation warning about space with operator string litteral

---------

Co-authored-by: Paul Baksic <30337881+bakpaul@users.noreply.github.com>
---
 Sofa/framework/Config/src/sofa/config.h.in | 4 ++--
 Sofa/framework/Type/src/sofa/type/Mat.h    | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Sofa/framework/Config/src/sofa/config.h.in b/Sofa/framework/Config/src/sofa/config.h.in
index a5e01a9ac1..2b2828c5fd 100644
--- a/Sofa/framework/Config/src/sofa/config.h.in
+++ b/Sofa/framework/Config/src/sofa/config.h.in
@@ -53,12 +53,12 @@ typedef double SReal;
 #endif
 
 /// User-defined literals allowing to convert any floating-point to a SReal
-constexpr SReal operator"" _sreal(long double real)
+constexpr SReal operator""_sreal(long double real)
 {
     return static_cast<SReal>(real);
 }
 /// User-defined literals allowing to convert any integer to a SReal
-constexpr SReal operator"" _sreal(unsigned long long int integer)
+constexpr SReal operator""_sreal(unsigned long long int integer)
 {
     return static_cast<SReal>(integer);
 }
diff --git a/Sofa/framework/Type/src/sofa/type/Mat.h b/Sofa/framework/Type/src/sofa/type/Mat.h
index e193389fdb..529cd97f09 100644
--- a/Sofa/framework/Type/src/sofa/type/Mat.h
+++ b/Sofa/framework/Type/src/sofa/type/Mat.h
@@ -28,6 +28,7 @@
 #include <sofa/type/Vec.h>
 
 #include <iostream>
+#include <algorithm>
 
 namespace // anonymous
 {
-- 
2.34.1

